### PR TITLE
Support EIP-6963: Multi Injected Provider Discovery

### PR DESCRIPTION
**Pull request by lochie**
**[EIP-6963](https://eips.ethereum.org/EIPS/eip-6963): Multi Injected Provider Discovery**

The biggest issue that this PR resolves is where multiple wallets try to override the `window.ethereum` object resulting in conflicts with one another.

This update to ConnectKit abstracts away all the new EIP-6963 wallet discovery logic into a useful `useWallets` Hook for easier future feature development.

**Preview:** https://connectkit-nextjs-git-eip-6963-avaraxyz.vercel.app/

---

**UX Logic:**

EIP-6963 enabled wallets are hoisted to the top of the discovery list, whilst legacy implementations using the `window.ethereum` object are left at the bottom to encourage wallets to update to the latest protocol.

---

**TODO:**

- [x] Remove unnecessary duplicated code and merge `WalletConfigProps` & `WalletProps` types.
   - Instead of removing these config files, we've instead just renamed the old EIP-1193 configs to be called `LegacyWalletProps` to continue supporting this protocol until EIP-6963 has been fully adopted.

### CODE REVIEW

**Comment by jxom** [comment]
i dedicate this song to you

https://www.youtube.com/watch?v=7aekxC_monc

**Comment by lochie** [comment]
> i dedicate this song to you
> 
> https://www.youtube.com/watch?v=7aekxC_monc

@jxom jamming to this while i jam on this PR ❤️

File: packages/connectkit/src/utils/midp.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { createStore } from 'mipd';
+
+export const createMidp = () => {
```
Comment (jxom) [inline packages/connectkit/src/utils/midp.ts:3]:
```suggestion
export const createMipd = () => {
```

Comment (lochie) [reply packages/connectkit/src/utils/midp.ts:3]: lol whoops. will update. thanks ❤️

Comment (lochie) [reply packages/connectkit/src/utils/midp.ts]: Fixed https://github.com/family/connectkit/pull/312/commits/9e00f0d665609d239ce6d2c5cb87b94094eae07a
Good catch @jxom, thanks for the review 🙌

**Comment by Songkeys** [comment]
Is this PR going to bring `wagmi@2` in?

**Comment by lochie** [comment]
> Is this PR going to bring `wagmi@2` in?

We're looking at bumping to wagmi@2 after this PR is live

**Comment by backseats** [comment]
To be clear, this resolves the issue where the user has phantom and metamask installed and active and the user goes to connect their metamask and sees a screen from metamask as if they don't have an account, right?

**Comment by lochie** [comment]
> To be clear, this resolves the issue where the user has phantom and metamask installed and active and the user goes to connect their metamask and sees a screen from metamask as if they don't have an account, right?

that is correct. thankfully EIP-6963 has been introduced to stop these issues 🙏

**Comment by backseats** [comment]
Thanks @lochie 

What's blocking this from merge and deploy? The actual EIP?

**Comment by backseats** [comment]
Amazing. Thanks!

**Comment by lochie** [comment]
lgtm

**Pull request merged**